exercise-app.js

```diff
@@ -28,13 +19,59 @@ const Weather = ({ temp, description, icon }) => (
 );
 
 
-const App = () => (
+class App extends React.Component {
+
+  constructor(props) {                // where do we hook up these props to ? where does this prop data come from?
```

Comment:
Stateful components are defined a [`class`](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Classes). JS classes have a special property called `constructor`. This function is called when you create a new instance of a class. You'll notice that `App` extends `React.Component`. So this function will receive props from its parent component. This is similar to `function Component(props) { ... }` but, using the class syntax instead of a function.